File: backend/models/Seccion.py
from __future__ import annotations
from pydantic import BaseModel
from datetime import datetime


class Seccion(BaseModel):
    nrc: int
    seccion: int
    titulo: str
    cuposMaximos: int
    cuposTomados: int
    modalidad: str
    fechaInicio: datetime
    fechaFin: datetime
    curso: Curso
    profesores: list[Profesor] = []
    horarios: list[BloqueTiempo] = []

    
    """ def __init__(self, rawDict: dict, cursos: dict, profs: dict)->None:
        self.seccion = rawDict.get("section")
        self.titulo = rawDict.get("title")
        self.cuposMaximos = rawDict.get("maxenrol")
        self.cuposTomados = rawDict.get("enrolled")
        self.modalidad = rawDict.get("campus")

        formato_fecha = "%Y-%m-%d %H:%M:%S"
        self.fechaInicio = datetime.strptime(rawDict.get("date_ini"))
        self.fechaFin = datetime.strptime(rawDict.get("date_fin"))

        # supone que ya existe el curso para esta materia
        self.curso = cursos.get(rawDict.get("class") + rawDict.get("course"))

        # supone que ya existen los profesores para esta materia
        for p in rawDict.get("instructors"):
            self.profesores.append(profs.get(p.get("name")))
        
        horarios = BloqueTiempo(rawDict.get("schedules")) """


# importaciones inferiores para evitar dependencias circulares
from models.Curso import Curso
from models.Profesor import Profesor
from models.BloqueTiempo import BloqueTiempo

Seccion.model_rebuild()

File: src/models/Seccion.ts
import Curso from './Curso';
import Profesor from './Profesor';
import BloqueTiempo from './BloqueTiempo';

export default  class Seccion {
  nrc: number;
  seccion: string;
  titulo: string;
  cuposMaximos: number;
  cuposTomados: number;
  modalidad: string;
  fechaInicio: Date;
  fechaFin: Date;
  curso: Curso;
  profesores: Profesor[] = [];
  horarios: BloqueTiempo[] = [];
  periodo: string;
  duracion: string;
  // Identificador para poder diferenciar secciones con el mismo NRC pero en diferentes períodos
  idUnico: string; 

  constructor(nrc: number, seccion: string, titulo: string, cuposMaximos: number, cuposTomados: number, modalidad: string, fechaInicio: Date, fechaFin: Date, curso: Curso, duracion:string, periodo: string, profesores: Profesor[] = [], horarios: BloqueTiempo[] = []) {
    this.nrc = nrc;
    this.seccion = seccion;
    this.titulo = titulo;
    this.cuposMaximos = cuposMaximos;
    this.cuposTomados = cuposTomados;
    this.modalidad = modalidad;
    this.fechaInicio = fechaInicio;
    this.fechaFin = fechaFin;
    this.curso = curso;
    this.profesores = profesores;
    this.horarios = horarios;
    this.duracion = duracion;
    this.periodo = periodo;
    this.idUnico = `${nrc}_${periodo}_${seccion}`;
  }
}
